src/Sylius/Behat/Page/Admin/Country/CreatePage.php

```diff
@@ -11,14 +11,83 @@
 
 namespace Sylius\Behat\Page\Admin\Country;
 
+use Behat\Mink\Driver\Selenium2Driver;
+use Behat\Mink\Element\Element;
 use Sylius\Behat\Behaviour\ChoosesName;
 use Sylius\Behat\Page\Admin\Crud\CreatePage as BaseCreatePage;
 
 /**
  * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
  * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
  */
 class CreatePage extends BaseCreatePage implements CreatePageInterface
 {
     use ChoosesName;
+
+    /**
+     * @var array
+     */
+    protected $elements = [
+        'provinces' => '#sylius_country_provinces',
+    ];
+
+    /**
+     * If the scenario is using a "@javascript" tag we are selecting Country name differently
+     *
+     * @param string $name
+     */
+    public function chooseNameProperly($name)
+    {
+        if ($this->getSession()->getDriver() instanceof Selenium2Driver) {
+            $this->chooseNameWithJavascript($name);
+
+            return;
+        }
+
+        $this->chooseName($name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fillProvinceData($name, $code, $abbreviation = null)
+    {
+        $this->getDocument()->clickLink('Add province');
+
+        $provinces = $this->getElement('provinces');
+
+        $provinceForm = $this->getLastProvinceElement();
+        $provincesCount = $provinceForm->getAttribute('data-form-collection-index');
+
+        $provinces->fillField('sylius_country_provinces_'.$provincesCount.'_name', $name);
```

Comment:
Looks baaad. We should get the last collection element and operate on it. If I have remembered correctly, we can get the items filtering by `[data-collection-type="item"]` or something like that.